app/controllers/users_controller.rb

```diff
@@ -3,26 +3,47 @@ def new
   end
 
   def create
-    if User.find_by(email: params[:signup][:email])
-      flash[:notice] = "Account Already Exists"
-      redirect_to new_user_path
+    if User.find_by(new_email)
     else
       user = User.new(user_params)
       user.tenant_id = session[:tenant_id]
       if user.valid?
-        user.save
-        session[:user_id] = user.id
-        flash[:notice] = "Thank you for creating an account."
-        redirect_to root_path
+        create_user(user)
       else
-        flash[:notice] = "Please try again."
-        redirect_to new_user_path
+        invalid_user(user)
       end
     end
   end
 
   private
 
+  def duplicate_email
+    flash[:notice] = "Account Already Exists"
+    redirect_to new_user_path
+  end
+
+  def new_email
+    params.require(:signup).permit(:email)
+  end
+
+  def invalid_user(user)
```

Comment:
Unused method argument - `user`. If it's necessary, use `_` or `_user` as an argument name to indicate that it won't be used. You can also write as `invalid_user(*)` if you want the method to accept any arguments but don't care about them.